vignettes/sits.Rmd

```diff
@@ -349,6 +349,110 @@ point_cf.tb <- sits_ndvi_arima_filter(point.tb, apply_whit = FALSE)
+
+In sits, the package kohonen (referencia) is used as basis to run the SOM, after the training step, to separate the samples in groups is necessary to label the neurons. A cluster can be one neuron or a set of neurons that belongs to same category. In this step, a neuron is named using majority vote technique through of original label given from sample data set. As a result of this, a sample that was associated to a neuron receives the same label of it. In some cases, no samples is associated to a neuron, then, it receives the label "Noclass".
+
+```{r}
+data("samples_mt_9classes", package = "sits")
+
+```
+```{r}
+data.tb <- samples_mt_9classes
+```
+
+
+```{r}
+# clustering time series using SOM
+kohonen_cluster <-
+    sits::sits_kohonen(
+        data.tb,
+        bands = c("evi", "ndvi"),
+        grid_xdim = 25,
+        grid_ydim = 25,
+        rlen = 100,
+        distance = "euclidean",
+        neighbourhood.fct = "gaussian",
+        mode = "online"
+
+    )
+# return a tibble with all samples and which cluster it belongs
+samples_cluster <- kohonen_cluster$info_samples
+samples_cluster [1:10,]
+```
+
+To verify the quality of clusters generated by SOM, the confusion matrix can be accessed. From the confusion matrix percentage of mixture within a cluster is shown. 
+
+```{r}
+
+# Calculate the percentage of confusion within a cluster 
+confusion_by_cluster <- sits_evaluate_cluster(kohonen_cluster$info_samples)
+confusion_by_cluster$mixture_cluster[1:5,]
+
+```
+
+```{r}
+# confusion matrix
+confusion_matrix <- confusion_by_cluster$confusion_matrix
+
+sits_plot_cluster_info(confusion_by_cluster, "Confusion by Cluster")
+```
+
+
+Besides that, to evaluate the samples with greater reliability, the process of clustering and labelling can be executed T times. 
+The process of clustering samples with SOM can be performed several times $t$ to evaluate the reliability of each sample. During this procedure, a historical table is created in order to store informations about sample and the cluster which it was associated at step $t$ %.
+The historical table contains for each sample the fields  identifier, the original label of a sample, given from sample dataset, the identifier and  label of neuron whose the sample was associated, and finally, the number of iteration in the clustering step.
+
+From historical table, metrics can be extracted to compute the probabilities of a sample belongs to a cluster. The contingency table for each sample is used to summarize the news labels associated to it, thus a new table summarized is created containing the fields the number of time that a sample $s_i$ was associated to a new label and the probabilities in percentage of a sample belongs to a cluster, identifier and original label of sample. From this metrics, the user can make decisions like to remove samples of dataset in order to improve the classification. 
```

Comment:
probability of a sample belonging.
The second sentence is too large.
From these metrics.
remove (bad?) samples FROM the dataset